migrator/src/main/java/com/google/phonenumbers/migrator/MigrationJob.java

```diff
@@ -13,23 +13,29 @@
 import java.nio.file.Paths;
 import java.util.ArrayList;
 import java.util.Arrays;
+import java.util.Collection;
 import java.util.List;
+import java.util.Map;
 import java.util.Scanner;
 
 /**
  * Represents a migration operation for a given region where each {@link MigrationJob} contains
- * a {@link RangeTree} of E.164 numbers to be migrated as well as the {@link CsvTable} which will
- * hold the available recipes that can be performed on the range. Only recipes from the given
- * two digit BCP-47 regionCode will be used.
+ * a map of E.164 numbers to be migrated as well as the {@link CsvTable} which will
+ * hold the available recipes that can be performed on the range. The number range map is a key value
+ * pair of the E.164 {@link RangeSpecification} representation of a number along with the raw input
+ * String originally entered. Only recipes from the given two digit BCP-47 regionCode will be used.
  */
 public final class MigrationJob {
 
   private final CsvTable<RangeKey> recipesTable;
-  private final RangeTree numberRange;
+  private final Map<RangeSpecification, String> numberRangeMap;
```

Comment:
Assuming this doesn't change, it might be best to make this immutable to make it clear that it will never change.